icu4c/source/data/unidata/generate.sh

```diff
@@ -30,6 +30,9 @@ rm $ICU4C_DATA_IN/coll/*.icu
 # Exit this shell script when a command fails.
 set -e
 
+# Workaround for obscure bug with the Bazel cache (ICU-22193).
+bazelisk clean --expunge
```

Comment:
Interesting! Since this change works, then this tells us what the problem is: the compiled code in the cache from the tools in Ubuntu 20.04 is interfering with the code compilation with the tools in Ubutunu 22.04.

Of course, force-clearing the cache cancels out any benefit of caching in the first place... 

My other comment in the Github Actions suggests a possibility in which we can keep the cache mechanism (if it works, then it should have some benefit) by tying it to the OS version.

If fixing the caching works, then we wouldn't need to always clear the cache first.

Reply:
I had simply assumed that this was run hermetically, the idea that the cache might be shared between different operating system versions didn't even cross my mind.